docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -54,27 +71,43 @@ VSOutput Main(VSInput input)
+First off, we define 2 types, `VSInput` and `VSOutput` which represent the 
+vertex shader's input and output.
+
+The input is 2, `float3` (vector of 3 float components), the first is the 
+"position" field, which are coordinates ranging from [-1.0, 1.0], if the values 
+are outside this range they are clipped, and we won't see them on screen.
 
-The input is 2, `float3` (vector of 3 float components), the first is the "position" field, which are coordinates ranging from [-1.0, 1.0], if the values are outside this range they are clipped, and we won't see them on screen.
+The second is the "color" field, which we also pass as the output of this stage 
+onto the pixel shader.
 
-The second is the "color" field, which we also pass as the output of this stage onto the pixel shader.
+Notice how all our fields have a colon and some identifier attached to them, 
+these are "semantics". Semantics that are preceded by `SV` are called 
+"system-value semantics" and their meaning and usage is defined by D3D11. 
+`SV_Position` for example means that the field `position` will be used by 
+D3D11 as the actual output of the vertex shader.
 
-Notice how all our fields have a colon and some identifier attached to them, these are "semantics".
-Semantics that are preceded by `SV` are called "system-value semantics" and their meaning and usage is defined by D3D11. `SV_Position` for example means that the field `position` will be used by D3D11 as the actual output of the vertex shader.
-Everything else are "user defined semantics" and their naming is up to us. These are used to pass data between shader stages.
+Everything else are "user defined semantics" and their naming is up to us. 
+These are used to pass data between shader stages.
 
-Then we have our `VSOutput`, which has our vertices in the first field `position` and our color in the second field `color`.
+Then we have our `VSOutput`, which has our vertices in the first field `position` 
+and our color in the second field `color`.
 
-Finally, we have our main function, which takes in a single parameter which is our input in the form of `VSInput`, and returns our output in the form of a `VSOutput`. Since we don't do any processing, we simply make a new instance of `VSOutput`, initialize it all to 0 and forward our input position and color to the output.
+Finally, we have our main function, which takes in a single parameter which is 
+our input in the form of `VSInput`, and returns our output in the form of a `VSOutput`.
+ Since we don't do any processing, we simply make a new instance of `VSOutput`, 
+ initialize it all to 0 and forward our input position and color to the output.
 
 ### Pixel Shader
 
-The **Pixel Shader** is the stage where we give the pixels on our render target color, it is invoked for each pixel that is covered by a triangle.
+The **Pixel Shader** is the stage where we give the pixels on our render target 
+color, it is invoked for each pixel that is covered by a triangle.
```

Comment:
being nitpicky on my side here, but this isn't 100% accurate since the GPU works in at least 2x2 tiles, so even if a pixel isn't covered by the rasterized triangle, it could still run a PS for it.., just covering bases here, might be worth mentioning, though I guess also rather advanced knowledge?

Reply:
its important information, if you make a RW buffer in a pixel shader its important to know it will be a block, not a 'smooth' triangle for what is calculated